### PR TITLE
Backport Bitcoin Qt/Gui changes up to 0.14.x part 1

### PR DESCRIPTION
**Pull request by codablock**
This is the first PR in a series of PRs to backport all Qt/Gui changes of Bitcoin up to 0.14.x.

The PRs were chosen by running:
```
git log --merges bitcoin/0.14 ^v0.12.2.x -- src/qt > backport-qt.txt
```
And then cherry picked into my local branch with:
```
git cherry-pick -m1 <hash-of-bitcoin-pr>
```

I tried to stick with Qt/Gui related PRs only, but sometimes it was required to pull in additional commits to resolve merge conflicts and compilation errors.
Also, my method of gathering the required PRs missed some PRs for some reason that I currently don't understand. Commits which could be found with a simple `git log -- src/qt` were missing when running the above command.

Later, when all the PRs that I catched and backported are merged into Dash, I will go through all Bitcoin PRs/commits again omitting the filtering from the above commands (with some custom scripting instead). I'll create additional PRs with these missing PRs/commits then. This may also include many non-Qt/Gui related PRs.

The next step will be to run a full comparison between Dash and Bitcoin and look for all kinds of suspicions differences between Dash and Bitcoin 0.14. This should reveal the last missing PRs to backport.

My long-term target is to get as close as possible to the Bitcoin code again, only omitting stuff that is not desired in Dash (like Segwit code).

EDIT: forgot to add list of included PRs. Working on it now.

### CODE REVIEW

**Comment by codablock** [comment]
I've just seen that Github was smart enough to reference all original PRs in the commit list. Is this enough for review or should I still add a list to the description?

File: src/qt/transactiontablemodel.cpp [hunk 1]
```diff
@@ -634,6 +634,34 @@ QVariant TransactionTableModel::data(const QModelIndex &index, int role) const
         return QString::fromStdString(rec->hash.ToString());
     case TxHexRole:
         return priv->getTxHex(rec);
+    case TxPlainTextRole:
+        {
+            QString details;
+            QDateTime date = QDateTime::fromTime_t(static_cast<uint>(rec->time));
+            QString txLabel = walletModel->getAddressTableModel()->labelForAddress(QString::fromStdString(rec->address));
+
+            details.append(date.toString("M/d/yy HH:mm"));
```
Comment (UdjinM6) [inline src/qt/transactiontablemodel.cpp]: IMO, it's confusing that date format here doesn't match the one on tx tab. I would propose to make sure they match, i.e.:
```diff
diff --git a/src/qt/transactiontablemodel.cpp b/src/qt/transactiontablemodel.cpp
index 28d67933b..497e374f6 100644
--- a/src/qt/transactiontablemodel.cpp
+++ b/src/qt/transactiontablemodel.cpp
@@ -637,10 +637,9 @@ QVariant TransactionTableModel::data(const QModelIndex &index, int role) const
     case TxPlainTextRole:
         {
             QString details;
-            QDateTime date = QDateTime::fromTime_t(static_cast<uint>(rec->time));
             QString txLabel = walletModel->getAddressTableModel()->labelForAddress(QString::fromStdString(rec->address));
 
-            details.append(date.toString("M/d/yy HH:mm"));
+            details.append(formatTxDate(rec));
             details.append(" ");
             details.append(formatTxStatus(rec));
             details.append(". ");
```

Comment (codablock) [reply src/qt/transactiontablemodel.cpp]: Agree. Just checked formatTxDate and it uses the system locale, as it should be. Added a commit on top.

**Comment by codablock** [comment]
Btw, even though I might have missed some PRs, all PRs required up until the progress overlay are done in my local branch. As far as I could see, everything works as expected. After this PR, there are 51 more commits which I'd split up in multiple PRs.

You wrote in Slack that you plan to do feature freeze this week. Would you consider waiting for these things to be merged in as well?

**Comment by UdjinM6** [comment]
>Would you consider waiting for these things to be merged in as well?

Can't promise anything. If it's more or less trivial we can merge them I guess. We can always revert them back before the release if anything is wrong (and merge fixed one in later versions e.g. 0.12.2.1).

**Comment by codablock** [comment]
Well trivial if you look at the amount of changed lines...not really :D
But most changes are luckily local to the Qt/Gui part and well tested on Bitcoin as I'd assume. Also, the changes didn't interfere much with Dash specific code. The only thing that interfered much with Dash code was the fee handling in the Gui, that is something I left out for now as it would need some discussion.

**Pull request merged**